Tests/UnitTests/Identity/CustomerInfoManagerTests.swift

```diff
@@ -208,6 +208,20 @@ class CustomerInfoManagerTests: BaseCustomerInfoManagerTests {
         expect(self.mockBackend.invokedGetSubscriberDataCount) == 1
     }
 
+    @MainActor
+    func testFetchAndCacheCustomerInfoIfStaleExecutesOnWorkerThread() throws {
+        mockDeviceCache.stubbedIsCustomerInfoCacheStale = false
+
+        var invokedCompletion = false
+        customerInfoManager.fetchAndCacheCustomerInfoIfStale(appUserID: Self.appUserID,
+                                                             isAppBackgrounded: false) { _ in
+            invokedCompletion = true
+        }
+
+        expect(invokedCompletion).toEventually(beTrue())
+        expect(self.mockOperationDispatcher.invokedDispatchOnWorkerThread) == true
```

Comment:
Not a fan of this since it's not a very accurate check, other things could be invoking this method. 
I don't know of a much better way to test when using this implementation, though.

Reply:
throwing something crazy in here, and not sure it will work, but maybe modifying `dispatchOnWorkerThread` to this:

```
func dispatchOnWorkerThread(delay: Delay = .none, caller: String = #function, file: String = #file, line: Int = #line, block: @escaping @Sendable () -> Void) {
```

would let you check what has called `dispatchOnWorkerThread` in the mock?

Reply:
definitely an interesting approach, I'll take a look